### ISSUE TITLE
new api method: ``run``  as alias for run_via_binary

### ISSUE THREAD

**Issue by jscotka** [fedora-modularity/conu#181]
Hi.
would be nice to have just wrapper and defined it in image API, new method ``run``, what will just pass thru params to ``run_via_binary`` (maybe could be defined directly in image, and not redefined in each image subclass)
because I expect that more backends will support just one method how to invoke container from image.
like:
```
     def run(self, *args, **kwargs):
           self.run_via_binary(*args, **kwargs)
```
### Benefits
 * use method run for people who are not interested if via binary or API
 * shorter method name for invoking
 * one method for backends what does not allow for example calling  via some python API

**Comment by TomasTomecek** [comment]
I like the idea. Unfortunately every tool has different a CLI/API. This is actually closely related to https://github.com/fedora-modularity/conu/issues/72 https://github.com/fedora-modularity/conu/issues/108

We need to figure a generic metadata format and let every backend implement it, then this should be easy to implement.

**Comment by jscotka** [comment]
Yes, this is little bit tricky part, that's why it is easier, to use ``run`` as alias for ``run_via_binary`` and just in case some backend don't have binary interface and just have python API, make there alias to run_via_api.

Harder is, that implementation what I've shown does not allow to help developers with parameter suggestions via ``*args, **kwargs`` 

so probably still should be possible explicitly call them via ``run_via_binary`` or via ``run_via_api`` but some shotcut like this could exist, maybe in image api definition? ;-)
```
     def run(self, *args, **kwargs):
          try:
               self.run_via_binary(*args, **kwargs)
          exception AttributeError:
                self.run_via_api(*args, **kwargs)

**Comment by TomasTomecek** [comment]
I'm absolutely against doing
```python
    def run(self, *args, **kwargs):
        try:
            self.run_via_binary(*args, **kwargs)
        exception AttributeError:
            self.run_via_api(*args, **kwargs)
```

That would be confusing to users as hell. As we both concluded -- none of the tools have the same API, that's why we created conu. So we can have an uniform API to access different container engines. We need to develop such API to run containers first, then we can proceed with resolving this issue.

If your proposed solution works for you, great, put it in your code, but that doesn't mean it would work well for everyone else. We need to make conu generic, easy to use and documented.

**Comment by jscotka** [comment]
Okay, I understand this PoV. and I'm familiar to avoid similar solution. My example were just, an example that this solution could be very simple, but tricky  and not clear for users.

**Comment by jscotka** [comment]
@dhodovsk see this issue, and probably would be nice to  improve it with today's comments about this ``run`` method
